generators/E906LegacyGen/SQPrimaryParticleGen.C

```diff
@@ -437,13 +453,14 @@ int SQPrimaryParticleGen::read_config(const char *cfg_file)
  * Reference: G. Moerno et.al. Phys. Rev D43:2815-2836, 1991
  * 
  * The pT and theta distributions generated are affected by the value of `_DrellYanGen`.
+ * The formula of pTmaxSq is discussed in DocDB 9292.
  */
 bool SQPrimaryParticleGen::generateDimuon(double mass, double xF)
 {
     double pz = xF*(DPGEN::sqrts - mass*mass/DPGEN::sqrts)/2.;
     
-    double pTmaxSq = (DPGEN::s*DPGEN::s*(1. - xF*xF) - 2.*DPGEN::s*mass*mass + mass*mass*mass*mass)/DPGEN::s/4.;
-  
+    //double pTmaxSq = (DPGEN::s*DPGEN::s*(1. - xF*xF) - 2.*DPGEN::s*mass*mass + mass*mass*mass*mass)/DPGEN::s/4.;
+    double pTmaxSq = (DPGEN::s / 4) * (1 - mass*mass/DPGEN::s)*(1 - mass*mass/DPGEN::s) * (1 - xF*xF);
```

Comment:
I forgot to mention one important change.  Here I have changed the definition of `pTmaxSq` to follow the discussion made in E906.  Actually the previous definition was problematic in the E1039 code, since it sometimes became negative.

Reply:
Thank you for the change. I also encountered the unphysical pTSq values in the past.

Reply:
For future reference. It is true that the old definition of `pTmaxSq` will return negative values. But this is not a problem if a consistent definition of  kinematic variable. Using a consistent definition, when the previous definition of  `pTmaxSq` is negative, xT and/or xB would be outside of [0,1].